prog/vm/github_runner.rb

```diff
@@ -166,16 +166,26 @@ def quota_available?
   label def wait_concurrency_limit
     unless quota_available?
       # check utilization, if it's high, wait for it to go down
-      utilization = VmHost.where(allocation_state: "accepting", arch: github_runner.label_data["arch"]).select_map {
-        sum(:used_cores) * 100.0 / sum(:total_cores)
-      }.first.to_f
+      utils = VmHost.where(allocation_state: "accepting", arch: github_runner.label_data["arch"])
+        .select_group(:family)
+        .select_append { round(sum(:used_cores) * 100.0 / sum(:total_cores), 2).cast(:float).as(:utilization) }
+        .to_hash(:family, :utilization)
 
-      unless utilization < 75
-        Clog.emit("Waiting for customer concurrency limit, utilization is high") { [github_runner, {utilization: utilization}] }
+      not_allow = if github_runner.label_data["arch"] == "x64" && github_runner.label_data["family"] == "standard" && github_runner.installation.premium_runner_enabled?
+        utils["premium"] > 75 && utils["standard"] > 75
```

Comment:
if there is either no `premium` or no `standard` host in `accepting` of the requested `arch`, then this would compare `nil` to 75 and fail.

Reply:
maybe use `utils` with `fetch` and a default value (0 or 100)?

Reply:
We shouldn't have this kind of issue in production since we need some hosts for each available runner label. However, it makes sense to use `.fetch(family, 0)` to be on the safe side. I also renamed it. Thanks for the review.

Reply:
I opted for `0` instead of `100` because if we don’t have that kind of machine, I prefer to stuck on `allocate_vm` instead of `wait_concurrency`.